google-cloud-tools-plugin/src/com/google/gct/idea/appengine/cloud/AppEngineDeploymentRunConfigurationEditor.java

```diff
@@ -129,15 +137,21 @@ public boolean value(VirtualFile file) {
           public File get() {
             return appEngineHelper.defaultAppYaml();
           }
-        }, appYamlPathField));
+        }, appYamlPathField, userSpecifiedArtifactFileSelector));
     generateDockerfileButton.addActionListener(
         new GenerateConfigActionListener(project, "Dockerfile", new Supplier<File>() {
           @Override
           public File get() {
+            File deploymentFile;
+            if (deploymentSource.getType() instanceof UserSpecifiedPathDeploymentSourceType) {
+              deploymentFile = new File(userSpecifiedArtifactFileSelector.getText());
+            } else {
+              deploymentFile = deploymentSource.getFile();
+            }
```

Comment:
Did you base this PR off of an intermediate stage of #565? I think I have since removed this code. Or maybe you just ended up doing the same thing (if you look at #565 this checking of the type is no longer needed).